packages/expo-updates/scripts/createFingerprint.js

```diff
@@ -0,0 +1,64 @@
+const { getConfig } = require('@expo/config');
+const Fingerprint = require('@expo/fingerprint');
+const fs = require('fs');
+const path = require('path');
+
+function findUpProjectRoot(cwd) {
+  if (['.', path.sep].includes(cwd)) return null;
+
+  if (fs.existsSync(path.join(cwd, 'package.json'))) {
+    return cwd;
+  } else {
+    return findUpProjectRoot(path.dirname(cwd));
+  }
+}
+
+(async function () {
+  const platform = process.argv[2];
+  const possibleProjectRoot = findUpProjectRoot(process.argv[3]);
+  const destinationDir = process.argv[4];
+
+  // Remove projectRoot validation when we no longer support React Native <= 62
+  let projectRoot;
+  if (fs.existsSync(path.join(possibleProjectRoot, 'package.json'))) {
+    projectRoot = possibleProjectRoot;
+  } else if (fs.existsSync(path.join(possibleProjectRoot, '..', 'package.json'))) {
+    projectRoot = path.resolve(possibleProjectRoot, '..');
+  }
+
+  process.chdir(projectRoot);
+
+  const { exp: config } = getConfig(projectRoot, {
+    isPublicConfig: true,
+    skipSDKVersionRequirement: true,
+  });
+
+  const runtimeVersion = config[platform]?.runtimeVersion ?? config.runtimeVersion;
+  if (!runtimeVersion || typeof runtimeVersion === 'string') {
+    return;
+  }
+
+  let fingerprint;
+  if (runtimeVersion.policy === 'fingerprintNativeExperimental') {
+    fingerprint = await Fingerprint.createProjectHashAsync(projectRoot, {
+      platforms: [platform],
+    });
+  } else if (runtimeVersion.policy === 'fingerprintNonNativeExperimental') {
+    // ignore everything in native directories to ensure fingerprint is the same
+    // no matter whether project has been prebuilt
+    fingerprint = await Fingerprint.createProjectHashAsync(projectRoot, {
+      platforms: [platform],
+      ignorePaths: ['/android/**/*', '/ios/**/*'],
+    });
```

Comment:
IIRC the fingerprint package has a `reasons` JSON output to explain how the hash was created (it lists all the dependencies it creates the hash over). People will want to publish an update, targeting a build in the appstore. If the update fingerprint is different from the build fingerprint, we should make it easy for them to see why the fingerprint was different. We should consider saving the `reasons` JSON too